### DISCUSSION TITLE
When is Candy actually going to be usable?

### DISCUSSION THREAD

**Discussion by JayteeNZ** [getcandy/candy-api#408] [Ideas]
I've been following Candy for a while, and yet, not once have I successfully managed to get a running version going without some sort of bullshit error.

What's the plan? It's all great that the package is supporting the latest Laravel versions etc, but the fucking thing doesn't even work as a core package.

I've been wanting to use this in a new app, with all risks involved, but honestly, it's starting to not be worth my time, and a custom build would be better at this rate.

This package has potential, and it's one of the most advanced ones out there, but the support is let's be honest, shit. Stop changing shit that doesn't need to be changed, and focus on actually getting a working product???

Thoughts?

**Comment by glennjacobs** [comment]
Hi Jack, thanks for the message. Before I start my reply, I would politely ask that we keep messages professional and not use profanity in the future. 

OK, let's address your concerns and questions...

## Can't get it working
We have lots of people running GetCandy, however, I totally appreciate it's complex and far less than perfect. We need to improve this. If you need support, please do jump on Discord where we will do our best to assist and are normally pretty responsive. If there are any specific issues you come across, please do raise an issue and if possible, offer a pull request to solve it.

## Support
Discord I would say is generally good, however, we all know the documentation is very poor and that makes or breaks a project. We've got to address this ASAP.

## What's the plan?
Annoyingly, we are understaffed at the moment which has delayed progress on this project. Extremely frustrating, but we're working hard to resolve that.

The Nuxt admin hub has been a problem for adoption. Although it works great, the set-up is too complex and lots of people struggle with getting Sanctum up and running etc, to the point we end up being Laravel Sanctum support it seems. We didn't want to throw it away and start again, but we did need a better solution. The decision has been made to try port the Nuxt admin hub to a Vue.js SPA that lives inside the Laravel app, which will simplify so many things and be more familiar to the Laravel devs out there, whilst offering exactly the same interface. It will also make plugin development a lot more straightforward, as a single package can supply both the API and admin hub functionality.

So the plan is to ...

1. Get the admin hub sorted once and for all
2. Document everything thoroughly

We won't be making any further changes until documentation is spot on.

**Reply by JayteeNZ** [reply]
Actually, before I read this, I'd like to apologize. I've been thinking about this comment I've made all day and was actually dreading a response.

Let's just say it's been one of those days in the office where things need to happen, and quickly.

However, I would like to contribute to this project, it has massive potential, and I'm happy to start reviewing the API and writing some documentation.

**Reply by glennjacobs** [reply]
Thanks @JayteeNZ, I appreciate that. We all get frustrated from time to time.

If you're able to jump on Discord, perhaps we could chat some more?

**Reply by JayteeNZ** [reply]
In the end, I managed to get a copy running by modifying package code (obviously, not ideal, but I wanted to be able to see GetCandy for once haha).

I think making elasticsearch mandatory (or an external search provider) is not ideal. That was something I didn't want to setup just to demo, and its not even a service I, nor others use often. Unfortunately products and such depended on it, and would throw errors and not show data (even tho it was being created) due to me not having elasticsearch installed and running.

**Reply by glennjacobs** [reply]
Elasticsearch is quite important to the project, as it brings so much that MySQL simply cannot offer. We thought about making a real simple MySQL driver, but it would add a load of work for us which simply isn't needed.

We intend to use Elasticsearch for reporting/analytics also.

Elasticsearch is pretty easy to install with things like TakeOut https://github.com/tighten/takeout

Again, I think documentation is the key here, rather than dumbing down the project.

**Reply by Tofandel** [reply]
I was actually wondering why you were using nuxt for the admin, a simple SPA is enough, otherwise you can even have a look at the new shiny thing: Laravel Jetstream with Inertia
https://jetstream.laravel.com/2.x/stacks/inertia.html
It's pretty much a vue MPA with a backend router and no need for api because variables are passed on the fly, different approach than the usual full SPA with an API but for an admin panel I think it makes sense, and you won't have to mess around with sanctum, it's a big rewrite, but if you already plan to drop it, the conversion makes sense
